apps/mocksi-lite/content/Popup/CreateDemo/index.tsx

```diff
@@ -43,31 +43,24 @@ const CreateDemo = ({
 		}
 	}, [createForm]);
 
-	const handleCancelClick = (recordings?: Recording[]) => {
-		if (recordings) {
-			setRecordings(recordings || []);
-		}
-		setCreateForm(false);
-	};
-
-	if (createForm) return <Form onCancel={handleCancelClick} />;
+	if (createForm) return <Form onCancel={() => setCreateForm(false)} />;
```

Comment:
Remember that recordings will sometimes be empty, like when doing a fresh install.

Please add this to prevent issues:
```suggestion
	if (createForm || !recordings.length )  {
		return <Form onCancel={handleCancelClick} />
	}
```

Reply:
mmm regarding the designs on figma, in the case we don't have any recordings we need to show the previous flow and hide the form
![image](https://github.com/Mocksi/HARlighter/assets/95246792/b335a33b-1417-4c9e-848b-df6c7013b8fe)

Reply:
Sounds good, you still need to return something (`null` in this case)
Otherwise you will get an error `Uncaught TypeError: recordings.map is not a function`
@nicolaschapur see https://app.rollbar.com/a/mocksi/fix/item/mocksi-lite/23

Reply:
I think that was an error when we do the JSON.parse of the recordings storage (in other cases it couldn't be null or undefined), so we where setting the state as undefined. I added a new function to get de recordings from storage with an error handler.